### PR TITLE
Format handlers consistently so that gencode can be applied

### PR DESCRIPTION
**Pull request by chloebrett**
This PR contains a bunch of preparatory work to make the handlers consistent, so that the code generator (which generates types, etc and is not in this PR) can be run.

Looks like a lot of line changes, but a lot of these are just reshuffling files, such as handlers that needed to be split up or moved to different parts of the codebase.

Requirements:
* Handlers must be declared using `const` syntax
* They must have separately declared types in the same file with the same name
* Their arguments must not use destructuring or default parameters
* Every handler must have its own file
* All files in the `main/handlers` folder will be exposed via IPC
* Each handler must be the default export in its file

These requirements are all documented in the gencode readme which is part of the gencode PR.

Other changes:
* Instead of simply injecting the `mainWindow`, handlers can access an `ipcContext` object which is passed in from `main/main.ts`
* 'START' and 'END' comments tell gencode where to inject generated code
* There is no `handlers/index` anymore

@hhen0005 and @Taz17 you two are the ones I'll be tagging in the gencode PR, so I've tagged you here as well.

### CODE REVIEW

**Comment by ghost** [comment]
preProcess tests are failing on this branch. Working on develop. Is that intended?

File: src/main/main.ts
Comment (ghost) [inline src/main/main.ts]: Does `IpcContext` need to be imported here?

**Comment by ghost** [comment]
Added @manikag-12 as this PR will affect her current export-file-menu PR

**Comment by chloebrett** [comment]
> preProcess tests are failing on this branch. Working on develop. Is that intended?

Hm no, will investigate

> Added @manikag-12 as this PR will affect her current export-file-menu PR

Sure, happy to do the rework myself if hers gets merged first

File: src/main/main.ts
Comment (chloebrett) [reply src/main/main.ts]: Yup, done thanks

**Review by chloebrett — COMMENTED** [review]
Fixed the test, good catch. Was due to using app.getPath before app was ready

File: src/sharedUtils.ts
Comment (ghost) [inline src/sharedUtils.ts:22]: lol

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Great stuff! Approved

**Comment by chloebrett** [comment]
Will wait until at least one other approval before merging

**Review by MutazAshhab — APPROVED** [review]

**Review by MutazAshhab — CHANGES_REQUESTED** [review]
I just ran the tests and the App.test.tsx isnt passing.

image

**Comment by chloebrett** [comment]
@Taz17 please pull latest and try again

I did fix some errors since your comment though I think the error you found was from an older commit

**Review by MutazAshhab — APPROVED** [review]
It passes all test cases now!

**Pull request merged**
